ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/runs/RunsTable.tsx

```diff
@@ -25,14 +27,23 @@ export type Run = {
   endedAt: string;
 };
 
-type RunsTableProps = {
+type RunsTableProps<TData> = {
   data: Run[] | undefined;
   sorting?: SortingState;
   setSorting?: OnChangeFn<SortingState>;
   isLoading?: boolean;
+  renderSubComponent: (props: { id: string }) => React.ReactElement;
+  getRowCanExpand: (row: Row<TData>) => boolean;
```

Comment:
Is a generic necessary here? Seems like the row type is always `Run`

```suggestion
type RunsTableProps = {
  data: Run[] | undefined;
  sorting?: SortingState;
  setSorting?: OnChangeFn<SortingState>;
  isLoading?: boolean;
  renderSubComponent: (props: { id: string }) => React.ReactElement;
  getRowCanExpand: (row: Row<Run>) => boolean;
```